src/BetterAppListTrait.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\apigee_edge_ui;
+
+/**
+ * Copyright (C) 2020 PRONOVIX GROUP BVBA.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+ * USA.
+ */
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\Core\Template\Attribute;
+use Drupal\apigee_edge\Entity\AppInterface;
+
+/**
+ * List builder additions for apps.
+ */
+trait BetterAppListTrait {
+
+  /**
+   * Alters the app info in the original list render array.
+   *
+   * @param array $build
+   *   The original render array.
+   * @param string $rel
+   *   The app name's link relationship type, defaults to 'canonical'.
+   */
+  private function buildAppListContent(array &$build, string $rel = 'canonical'): void {
+    // Use custom template instead of table.
+    unset($build['table']['#type']);
+    if (isset($this->entityTypeId)) {
+      $build['table']['#type'] = $this->entityTypeId;
+    }
+    $build['table']['#theme'] = 'apigee_edge_ui_list';
+
+    // Build the app rows from scratch.
+    $build['table']['#items'] = [];
+    foreach ($this->load() as $entity) {
+      /** @var \Drupal\apigee_edge\Entity\AppInterface $entity */
+      $app_row = [
+        '#attributes' => new Attribute([
+          'id' => $this->getCssIdForInfoRow($entity),
```

Comment:
Ohh wow, now I see why `AppListBuilder` is extended... a trait must not have dependencies to properties/methods and anything else that is not defined in the trait or required by a trait (f.ex with abstract methods).
The simplest solution is instead of using a trait to override default developer/team app UIs is defining your own base class (either by testing the vanilla AppListBuilder or not) and then defining your own developer- and team app-specific subclasses of that class.